### DISCUSSION TITLE
Acquiring BNC input from pycontrol board

### DISCUSSION THREAD

**Discussion by msoheib** [bonsai-rx/bonsai#1388] [Q&A]
Dear All,

I have been trying to obtain signals from a pycontrol board into bonsai-rx using a serial micro cable. I tried to create a serial port node in Bonsai-rx, but I was not able to get any readings with DC input. Outside bonsai I am able to do this via mpremote python library but would be nice to work within Bonsai. Is there a firmware similar to firmata for ardiuno for micropython that can help with direct acquisition within bonsai?

thank you.

**Comment by bruno-f-cruz** [comment]
Although pycontrol seems to communicate via a generic serial connection, it implements its own protocol. Bonsai serial nodes are generic, in that that they are able to open connections with most serial devices. Unfortunately, Bonsai knows nothing about the specifics of the protocol pycontrol implements so, unless I am missing something, I don't think you will get anything super useful by just opening the port. For instance, I suspect you might need to issue specific instructions to even handshake with the pycontrol board and start getting messages through.

I think you might have better luck [asking in their discussions directly instead](https://github.com/orgs/pyControl/discussions).
